lib/std/fs/test.zig

```diff
@@ -1404,3 +1404,69 @@ test "File.PermissionsUnix" {
     try testing.expect(permissions_unix.unixHas(.user, .execute));
     try testing.expect(!permissions_unix.unixHas(.other, .execute));
 }
+
+test "preopens" {
+    if (builtin.os.tag != .wasi) return error.SkipZigTest;
+
+    // lifted from `testing`
+    const random_bytes_count = comptime (12);
+    const sub_path_len = comptime (std.fs.base64_encoder.calcSize(random_bytes_count));
+    const path = comptime ("/tmp");
+    const tmp_file_name = comptime ("file.txt");
+    const nonsense = comptime ("nonsense");
+
+    var random_bytes: [random_bytes_count]u8 = undefined;
+    var sub_path: [sub_path_len]u8 = undefined;
+
+    std.crypto.random.bytes(&random_bytes);
+    _ = std.fs.base64_encoder.encode(&sub_path, &random_bytes);
```

Comment:
Ideally, you use the result of this function, rather than the buffer it writes to.

Reply:
in my "defense" this is lifted pretty much verbatim from `testing.tmpDir` and `testing.tmpIterableDir` https://github.com/ziglang/zig/blob/master/lib/std/testing.zig#L545-L548 maybe I will contribute (separately) an improvement there too then :)